### DISCUSSION TITLE
Unique validation dynamic ignore

### DISCUSSION THREAD

**Discussion by horo42** [spatie/laravel-data#8] [Q&A] [answered]
When having a unique constraint on a user data objects email field, how would you ignore the currently edited user models email?

**Comment by rubenvanassche** [comment, marked as answer]
Hi @horo42,

Good point! I've added support to the `Unique` rule to support closures so you can read the request:

https://github.com/spatie/laravel-data/blob/912c51b5812838391d4d779bdeb8984af8c00d6e/docs/advanced-usage/validation-attributes.md#L855

**Reply by ansien** [reply]
Looks cool, but in what world is that valid syntax?

![image](https://user-images.githubusercontent.com/1700132/140184198-3ab13c6b-2947-4482-be1d-0d1ed254ad18.png)

**Reply by rubenvanassche** [reply]
Hmm, it is actually not allowed to use closures within attributes, my bad! In that case, I would add the rule to the static `rules` method, it seems that we cannot support this at this moment.

**Reply by ansien** [reply]
Yeah that's how I ended up doing it too. Hopefully they'll add support for closures in attributes in a future PHP version.

**Comment by dineshram7660** [comment]
I tried this like
```
#[Unique('table_name', ignore: request()->get('column_name'))]
```
But the constructor gives an error `Constant expression contains invalid operations`.

What should i do?

**Reply by ansien** [reply]
Use the static rules method
